### ISSUE TITLE
Authenticate silently

### ISSUE THREAD

**Issue by teferio** [box/box-android-sdk#417]
Hello,

I am using the box-android-sdk and I was looking for a way to let users view files without showing the Login Screen.

I believe this is possible through the OAuth2.0 JWT authentication. If I understand correctly, once the config.json file is generated, the app can use that to authenticate silently, without showing the Login Screen. **Given that the users shouldn't necessarily be box-users.**

But I was only able to find the implementation of this in the **box-java-sdk** not in **box-android-sdk**.
https://github.com/box/box-java-sdk/blob/master/doc/authentication.md#server-authentication-with-jwt

Is there a way to authenticate silently the same way with JWT config.json in the box-android-sdk?

Thanks in advance.

**Comment by doncung** [comment]
Jwt logic is meant to be done server side to ensure security for the token.
The purpose for jwt is for customers to manage their own users so you might
create your own login page for instance and tie it to your own
authentication mechanisms.

**Comment by teferio** [comment]
Hello doncung,

Thanks for the Jwt info. I was more concerned with authenticating non-box users silently. Let me rephrase my question to a more clearer sense:

Is there a way in the box API to do silent authentication?

**silent authentication**: authenticating non-box users without login screen, using some kind of api token, hardcoded in the android app.

**Comment by doncung** [comment]
You could do that through jwt, but it would require your own server
endpoint to associate a box user to one of your users.

**Comment by teferio** [comment]
Thanks doncung,

It's helpful to know that I need my own server to do it through JWT.
I will do that on my own server then! But to connect it to my android app, do I have to use the **box-java-sdk**?

Couldn't I use the **box-android-sdk** instead?

I'm asking this because I only see samples of JWT authentication on the **box-java-sdk**
https://github.com/box/box-java-sdk/blob/master/doc/authentication.md#server-authentication-with-jwt

I don't see this available through the **box-android-sdk**

Thanks in advance.

**Comment by dblankety** [comment]
You would use the java sdk or another server side language sdk to implement
your server side jwt logic. That way your unique token can be preserved.
You would use the Android SDK with the refresh provider implementation to
call in to your server endpoint when users tokens need to be refreshed or
if its the first time and a user needs to be associated with the user.

**Comment by doncung** [comment]
The box-android-sdk supports an AuthenticationRefreshProvider interface which is designed to hook into your server. If you want to make it silent you would implement launchAuthUi to silently call your server to get a valid access token for your user.

**Comment by teferio** [comment]
Thanks!

**Comment by PJSimon** [comment]
Hi @teferio, 

It looks like this issue can be closed, but if you're still having issues, please don't hesitate to re-open it!

Thanks,

Patrick
